cmake/host-tools.cmake

```diff
@@ -32,18 +44,21 @@ else()
     ${item}\n\
   But the minimum supported version is ${MIN_WEST_VERSION}\n\
   Please upgrade with:\n\
-      pip3 install --upgrade west")
+      ${PYTHON_EXECUTABLE} -m pip install --upgrade west\
+  ${PYTHON_EXECUTABLE_OUT_OF_SYNC}")
   endif()
+
   # Just output information for a single version. This will still work
   # even after output is one line.
   message(STATUS "Found west: ${WEST} (found suitable version \"${west_version}\", minimum required is \"${MIN_WEST_VERSION}\")")
 
-    execute_process(
-      COMMAND ${WEST}  topdir
-      OUTPUT_VARIABLE  WEST_TOPDIR
-      OUTPUT_STRIP_TRAILING_WHITESPACE
-      WORKING_DIRECTORY ${ZEPHYR_BASE}
-      )
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} -c
+    "from west.app.main import main; main(['topdir'])"
```

Comment:
Same thing here; no need to expose the CLI implementation details. We can just use the west API directly.

```
>>> from west.util import west_topdir
>>> print(west_topdir())
/home/mbolivar/zp
```

I have no idea why I didn't think of this sooner; I'm so sorry for wasting your time.

Reply:
done